### PR TITLE
Output date string from DateFormatEnums 

### PR DESCRIPTION
**Pull request by jonnyh404**

### CODE REVIEW

**Review by obi-isebor — APPROVED** [review]
Looks good

File: ResSimpy/ISODateTime.py [hunk 1]
```diff
@@ -98,3 +98,40 @@ def convert_to_iso(cls: type[ISODateTime], date: str, date_format: Optional[Date
...
+        str: The converted date
+        """
+
+        # check if the iso time is non-zero
+        if self.time() != datetime.min.time():
+            time_included = True
+        else:
+            time_included = False
+
+        match new_date_format:
```
Comment (kwr3) [inline ResSimpy/ISODateTime.py:118]: nice use of switch statements

File: tests/Nexus/date_format_tests/test_iso_date_properties.py [hunk 2]
```diff
@@ -188,3 +188,21 @@ def test_datetime_to_iso():
 
     # Assert
     assert result == expected_iso_date
+
+
+@pytest.mark.maintain_datetime_behaviour
+@pytest.mark.parametrize('iso_date, expected_str, dateformat', [
+    (ISODateTime(2022, 1, 19), '19 JAN 2022', DateFormat.DD_MMM_YYYY),
```
Comment (kwr3) [inline tests/Nexus/date_format_tests/test_iso_date_properties.py:195]: tests look good

**Review by kwr3 — APPROVED** [review]
looks good

**Pull request merged**